lib/shopify-cli/context.rb

```diff
@@ -446,6 +453,23 @@ def which(cmd)
       nil
     end
 
+    # Checks if there's a newer version of the CLI available and returns version string if
+    # this should be conveyed to the user (i.e., if it's been over 24 hours since last check)
+    #
+    # #### Parameters
+    #
+    # #### Returns
+    # - `version`: string of newer version available, IFF new version is available AND it's time to inform user,
+    #            : nil otherwise
+    #
+    def new_version
+      if (time_of_last_check + VERSION_CHECK_INTERVAL) < (now = Time.now.to_i)
+        update_time_of_last_check(now)
+        latest_version = retrieve_latest_gem_version
+        latest_version unless latest_version == ShopifyCli::VERSION
```

Comment:
Would this mean it would show you the upstream version even if it was technically _older_ than the current version? I could imagine this tripping people up if they're using a beta version or something.

Reply:
Yes, it would!  However, given the highly unlikely situation of that occurring (more later), I think this quick/cleaner check is better than the alternative of doing string parsing to determine which is newer, or even engaging the SemVer library to make the check (extra complication for an edge case).

Since our current release strategy for all the different packages (brew, Debian, RPM, etc.) are all based upon the pre-existence of the RubyGems version, it's highly unlikely that any external user is going to have a version installed that is newer than the latest version on rubygems.org.  The likeliest scenario of that happening is with internal development, in which case `ctx.development?` will return `true` and the new version check won't occur anyway!